plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/MockThriftHttpMetastoreServer.java

```diff
@@ -0,0 +1,1763 @@
+
+        ServerConnector connector = new ServerConnector(jettyServer, new HttpConnectionFactory());
+        port = getRandomAvailablePort();
+        connector.setPort(port);
+        connector.setReuseAddress(true);
+        jettyServer.addConnector(connector);
+
+        MockThriftHiveMetastoreHandler mockThriftHandler = new MockThriftHiveMetastoreHandler(delegate);
+        TProcessor processor = new ThriftHiveMetastore.Processor<>(mockThriftHandler);
+        thriftHttpServlet = new MockThriftHttpServelet(processor, new TBinaryProtocol.Factory(), Optional.of(requestInterceptor));
+        ServletContextHandler context = new ServletContextHandler(
+                ServletContextHandler.NO_SECURITY | ServletContextHandler.NO_SESSIONS);
+        jettyServer.setHandler(context);
+
+        context.addServlet(new ServletHolder(thriftHttpServlet), httpPath);
+        start();
+    }
+
+    public void start()
+            throws Exception
+    {
+        jettyServer.start();
+    }
+
+    public void stop()
+            throws Exception
+    {
+        jettyServer.stop();
+    }
+
+    @ResourcePresence
+    public boolean isRunning()
+    {
+        return jettyServer.isRunning();
+    }
+
+    private int getRandomAvailablePort()
+            throws IOException
+    {
+        int freePort = -1;
+        try (ServerSocket temp = new ServerSocket(0)) {
+            freePort = temp.getLocalPort();
+        }
+        return freePort;
+    }
+
+    public int getPort()
+    {
+        return port;
+    }
+
+    private static class MockThriftHttpServelet
+            extends TServlet
+    {
+        private final Optional<TestHttpRequestInterceptor> requestInterceptor;
+
+        public MockThriftHttpServelet(
+                TProcessor processor,
+                TProtocolFactory protocolFactory,
+                Optional<TestHttpRequestInterceptor> requestInterceptor)
+        {
+            super(processor, protocolFactory);
+            this.requestInterceptor = requestInterceptor;
+        }
+
+        @Override
+        protected void doPost(HttpServletRequest request,
+                HttpServletResponse response)
+                throws ServletException, IOException
+        {
+            requestInterceptor.ifPresent(testHttpRequestInterceptor -> testHttpRequestInterceptor.process(request));
+            super.doPost(request, response);
+        }
+    }
+
+    interface TestHttpRequestInterceptor
+    {
+        void process(HttpServletRequest request);
+    }
+
+    static class MockThriftHiveMetastoreHandler
+            implements ThriftHiveMetastore.Iface
+    {
+        private final InMemoryThriftMetastore delegate;
+
+        public MockThriftHiveMetastoreHandler(InMemoryThriftMetastore delegate)
+        {
+            this.delegate = delegate;
+        }
+
+        @Override
+        public String getMetaConf(String key)
+                throws MetaException, TException
+        {
+            return null;
```

Comment:
We should make all of these un-implemented methods throw `UnsupportedOperationException`, to make it easier to debug if they are ever called.

Reply:
Makes sense.